### PR TITLE
feat(playground): add dataset selector and dataset examples table to playground

### LINKED ISSUE #5281: [playground][ui] add dataset selector and dataset examples table to playground
Add a clearable dataset selector to the playground. When a dataset s selected display a dataset examples table with input, reference output and output columns for each prompt

### PR DESCRIPTION
**Pull request by Parker-Stafford**
resolves #5281 

- adds dataset selector and rudimentary table with instance output to playground
- refactors input out of instances (was incorrectly duplicated)
- removes input mode and derives it from the presence of datasetId

https://github.com/user-attachments/assets/e3218f0e-e68b-4f64-b32b-756ad04756e4

saved vars

https://github.com/user-attachments/assets/311343f0-e99c-4649-afa5-4f51f50b4945

### CODE REVIEW

File: app/src/components/dataset/CreateDatasetForm.tsx [hunk 1]
```diff
@@ -45,6 +45,25 @@ export function CreateDatasetForm(props: CreateDatasetFormProps) {
         onCompleted: (response) => {
           onDatasetCreated(response["createDataset"]["dataset"]);
         },
+        updater: (store) => {
```
Comment (Parker-Stafford) [inline app/src/components/dataset/CreateDatasetForm.tsx:48]: finally got this working, not sure what hte problem was before. No longer need to do this https://github.com/Arize-ai/phoenix/issues/4008 which is flakey at best and can start to update connections in teh store directly this way. Can also remove fetch keys etc.

File: app/src/components/dataset/NewDatasetButton.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export function NewDatasetButton({
+  onDatasetCreated,
+}: {
+  onDatasetCreated: (datasetId: string) => void;
+}) {
+  const [error, setError] = useState<string | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const notifySuccess = useNotifySuccess();
+  return (
+    <PopoverTrigger
```
Comment (Parker-Stafford) [inline app/src/components/dataset/NewDatasetButton.tsx:26]: this is just pulled from the dataset picker that was coupled to this button on the add dataset to traces dialog

File: app/src/components/dataset/DatasetPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { graphql, useLazyLoadQuery } from "react-relay";
+
+import { Item, Picker, PickerProps } from "@arizeai/components";
+
+import { DatasetPickerQuery } from "./__generated__/DatasetPickerQuery.graphql";
+
+type DatasetPickerProps = Pick<
```
Comment (Parker-Stafford) [inline app/src/components/dataset/DatasetPicker.tsx:8]: this is just pulled from add traces to dataset dialog, decoupled from the button (see above) and has a  few props added

File: app/src/pages/playground/PlaygroundDatasetExamplesTable.tsx [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+  return (
+    <LargeTextWrap>
+      <JSONText json={value} space={2} />
+    </LargeTextWrap>
+  );
+}
+
+const PAGE_SIZE = 100;
+
+//un-memoized normal table body component - see memoized version below
```
Comment (Parker-Stafford) [inline app/src/pages/playground/PlaygroundDatasetExamplesTable.tsx:51]: this is for column resizing and copied from the experiment compare table which is copied from here https://tanstack.com/table/latest/docs/framework/react/examples/column-resizing-performant

File: app/src/pages/playground/PlaygroundDatasetExamplesTable.tsx [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+    [hasNext, isLoadingNext, loadNext]
+  );
+
+  /**
+   * Instead of calling `column.getSize()` on every render for every header
+   * and especially every data cell (very expensive),
+   * we will calculate all column sizes at once at the root table level in a useMemo
+   * and pass the column sizes down as CSS variables to the <table> element.
+   */
+  const columnSizeVars = React.useMemo(() => {
```
Comment (Parker-Stafford) [inline app/src/pages/playground/PlaygroundDatasetExamplesTable.tsx:207]: see above pulled from tanstack example

File: app/src/pages/playground/PlaygroundInputModeRadioGroup.tsx [hunk 6]
```diff
@@ -1,32 +0,0 @@
-import React from "react";
```
Comment (Parker-Stafford) [inline app/src/pages/playground/PlaygroundInputModeRadioGroup.tsx:1]: not needed if we derive state from selection of a dataset - this is the norm for other playgrounds

File: app/src/pages/trace/SpanToDatasetExampleDialog.tsx [hunk 7]
```diff
@@ -180,13 +179,28 @@ export function SpanToDatasetExampleDialog({
...
-                  />
+                  <Flex direction="row" gap="size-100" alignItems="end">
+                    <DatasetPicker
+                      onSelectionChange={onChange}
+                      onBlur={onBlur}
+                      validationState={invalid ? "invalid" : "valid"}
+                      errorMessage={error?.message}
+                      label="Dataset"
+                    />
+                    <NewDatasetButton
```
Comment (Parker-Stafford) [inline app/src/pages/trace/SpanToDatasetExampleDialog.tsx:190]: these are just pulled out of the same component and reusable now

File: js/examples/notebooks/tracing_openai_node_tutorial.ipynb [hunk 8]
```diff
@@ -62,7 +62,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 2,
+   "execution_count": null,
```
Comment (cephalization) [inline js/examples/notebooks/tracing_openai_node_tutorial.ipynb:65]: intentional?

File: app/src/components/dataset/DatasetPicker.tsx (hunk 3 above)
Comment (Parker-Stafford) [reply app/src/components/dataset/DatasetPicker.tsx:8]: ahh yes i w refactoring / moving in this case so didn't think of it but will do next time

File: js/examples/notebooks/tracing_openai_node_tutorial.ipynb (hunk 8 above)
Comment (Parker-Stafford) [reply js/examples/notebooks/tracing_openai_node_tutorial.ipynb:65]: just ran a notebook to get some spans in then this changed i think there's some clean tutorials script but this is just metadata on the notebook cel so fine to change

**Review by cephalization — APPROVED** [review]

**Pull request merged**